python/looker_sdk/rtl/api_settings.py

```diff
@@ -40,105 +38,125 @@
     from typing_extensions import Protocol
 
 
-def _convert_bool(val: str, _: bool) -> bool:
-    converted: bool
-    if val.lower() in ("yes", "y", "true", "t", "1"):
-        converted = True
-    elif val.lower() in ("", "no", "n", "false", "f", "0"):
-        converted = False
-    else:
-        raise TypeError
-    return converted
-
-
 class PApiSettings(transport.PTransportSettings, Protocol):
-    def get_client_id(self) -> Optional[str]:
+    looker_url: Optional[str]
+    redirect_uri: Optional[str]
+
+    @property
+    def client_id(self) -> Optional[str]:
         ...
 
-    def get_client_secret(self) -> Optional[str]:
+    @property
+    def client_secret(self) -> Optional[str]:
```

Comment:
@jkaster I wanted to have a conversation with you about what our settings interface should look like. I see that in typescript you have a `[key: string]: any` but I'm wondering if we should take the route of being more explicit in the interface enumerating all the possible settings? That way someone can implement they're own settings provider and not have to guess at what's in `[key: string]: any` (or read the code) in order to make their settings work for our different RTL components.

Let's chat about it

Reply:
Yeah, let's catch up on this. The reason the Typescript definition is so loose is because the configuration values someone may retrieve via (typically) the `readConfig()` overload could be values we know nothing about. Their authentication process could be completely different than any we've provided samples for, without us knowing what values they'd need. For example, maybe they're retrieving credentials from a key store on GCP, Azure, or AWS.

Reply:
recap on this conversation:
1/ instance properties/members will be the properties on `PTransportSettings`
2/ `readConfig` will produce a dictionary with keys corresponding to further settings that `AuthSession` needs
3/ nice to have: `AuthSession` should validate the necessary keys are present in `readConfig` (e.g. for api3 auth verify `client_id` and `client_secret` are returned, for oauth verify `client_id`, `looker_url`, and `redirect_uri` are returned)